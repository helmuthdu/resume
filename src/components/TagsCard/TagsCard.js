import React from 'react';
import { Card, List } from 'semantic-ui-react';

interface TagsCardProps {
  tags: Array<any>;
  Link: any;
  tag?: string;
}

export default (props: TagsCardProps) => {
  return (
    <Card>
      <Card.Content>
        <Card.Header>Tags</Card.Header>
      </Card.Content>
      <Card.Content>
        <List>
          {props.tags.map(tag => {
            const isActive = tag.fieldValue === props.tag;
            const activeStyle = {
              fontWeight: '700'
            };
            return (
              <List.Item as="a" key={tag.fieldValue}>
                <List.Icon name="tag" color={isActive ? 'blue' : null} />
                <List.Content style={isActive ? activeStyle : null}>
                  <props.Link to={`/blog/tags/${tag.fieldValue}/`}>
                    {tag.fieldValue} ({tag.totalCount})
                  </props.Link>
                </List.Content>
              </List.Item>
            );
          })}
        </List>
      </Card.Content>
    </Card>
  );
};
