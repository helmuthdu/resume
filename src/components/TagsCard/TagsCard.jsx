import PropTypes from 'prop-types';
import { Card, List } from 'semantic-ui-react';

const TagCard = props => {
  return (
    <Card>
      <Card.Content>
        <Card.Header>Tags</Card.Header>
      </Card.Content>
      <Card.Content>
        <List>
          {props.tags.map(tag => {
            const isActive = tag.fieldValue === props.tag;
            const activeStyle = {
              fontWeight: '700'
            };
            return (
              <List.Item as="a" key={tag.fieldValue}>
                <List.Icon name="tag" color={isActive ? 'blue' : null} />
                <List.Content style={isActive ? activeStyle : null}>
                  <props.Link to={`/blog/tags/${tag.fieldValue}/`}>
                    {tag.fieldValue} ({tag.totalCount})
                  </props.Link>
                </List.Content>
              </List.Item>
            );
          })}
        </List>
      </Card.Content>
    </Card>
  );
};

TagCard.propTypes = {
  tag: PropTypes.string.isRequired,
  tags: PropTypes.array.isRequired
};

export default TagCard;
