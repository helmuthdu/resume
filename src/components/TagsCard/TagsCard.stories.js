import { action } from '@storybook/addon-actions';
import { select, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import TagsCard from './TagsCard';

const withReadme = require('storybook-readme/with-readme').default;
const TagsCardReadme = require('./README.md');

const tags = [
  { fieldValue: 'tag01', totalCount: 2 },
  { fieldValue: 'tag02', totalCount: 4 },
  { fieldValue: 'tag03', totalCount: 6 }
];

const LinkStub = props => <div onClick={action(props.to.toString())}>{props.children}</div>;

storiesOf('TagsCard', module)
  .addDecorator(withReadme(TagsCardReadme))
  .addDecorator(withKnobs)
  .add('default', () => {
    return <TagsCard tags={tags} Link={LinkStub} />;
  })
  .add('with tag property', () => {
    const tag = select(
      'Tag',
      tags.map(t => t.fieldValue),
      'tag01'
    );
    return <TagsCard tags={tags} tag={tag} Link={LinkStub} />;
  });
