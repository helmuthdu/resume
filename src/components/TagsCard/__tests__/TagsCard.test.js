import { shallow } from 'enzyme';
import React from 'react';
import { List } from 'semantic-ui-react';
import TagsCard from '../TagsCard';

describe('component -> TagsCard', () => {
  let LinkStub;

  beforeEach(() => {
    LinkStub = props => <div>{props.children}</div>;
  });

  it('should list all the tags', () => {
    const tags = [
      { fieldValue: 'tag01', totalCount: 2 },
      { fieldValue: 'tag02', totalCount: 4 },
      { fieldValue: 'tag03', totalCount: 6 }
    ];

    const wrapper = shallow(<TagsCard tags={tags} Link={LinkStub} />);

    expect(wrapper.find(List.Item)).toHaveLength(3);
  });

  it('should have on tag active', () => {
    const tags = [
      { fieldValue: 'tag01', totalCount: 2 },
      { fieldValue: 'tag02', totalCount: 4 },
      { fieldValue: 'tag03', totalCount: 6 }
    ];

    const wrapper = shallow(<TagsCard tags={tags} Link={LinkStub} tag="tag01" />);

    expect(wrapper).toMatchSnapshot();
  });
});
