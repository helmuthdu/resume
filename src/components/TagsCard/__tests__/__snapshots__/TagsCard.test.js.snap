// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component -> TagsCard should have on tag active 1`] = `
<Card>
  <CardContent>
    <CardHeader>
      Tags
    </CardHeader>
  </CardContent>
  <CardContent>
    <List>
      <ListItem
        as="a"
        key="tag01"
      >
        <ListIcon
          color="blue"
          name="tag"
        />
        <ListContent
          style={
            Object {
              "fontWeight": "700",
            }
          }
        >
          <LinkStub
            to="/blog/tags/tag01/"
          >
            tag01
             (
            2
            )
          </LinkStub>
        </ListContent>
      </ListItem>
      <ListItem
        as="a"
        key="tag02"
      >
        <ListIcon
          color={null}
          name="tag"
        />
        <ListContent
          style={null}
        >
          <LinkStub
            to="/blog/tags/tag02/"
          >
            tag02
             (
            4
            )
          </LinkStub>
        </ListContent>
      </ListItem>
      <ListItem
        as="a"
        key="tag03"
      >
        <ListIcon
          color={null}
          name="tag"
        />
        <ListContent
          style={null}
        >
          <LinkStub
            to="/blog/tags/tag03/"
          >
            tag03
             (
            6
            )
          </LinkStub>
        </ListContent>
      </ListItem>
    </List>
  </CardContent>
</Card>
`;
